lib/QubitFindingAid.class.php

```diff
@@ -0,0 +1,460 @@
+     */
+    public function setHomeDir(?string $dir): void
+    {
+        if (isset($dir)) {
+            $this->homeDir = $dir;
+
+            return;
+        }
+
+        // Default finding aid home directory
+        $this->homeDir = sfConfig::get('sf_web_dir').'/downloads/';
+    }
+
+    /**
+     * Get the home directory for finding aid files.
+     *
+     * @param bool $absolute return absolute path if true (default), relative
+     *                       path if false
+     */
+    public function getHomeDir(?bool $absolute = true): string
+    {
+        // Return absolute (filesystem) path
+        if (true === $absolute) {
+            return $this->homeDir;
+        }
+
+        // Return relative path to file
+        return str_replace(sfConfig::get('sf_web_dir').'/', '', $this->homeDir);
+    }
+
+    /**
+     * Get possible file names for the finding aid.
+     *
+     * @return array of possible filenames
+     */
+    public function getPossibleFilenames(): array
+    {
+        $filenames = [
+            $this->resource->id.'.pdf',
+            $this->resource->id.'.rtf',
+        ];
+
+        if (null !== $this->resource->slug) {
+            $filenames[] = $this->resource->slug.'.pdf';
+            $filenames[] = $this->resource->slug.'.rtf';
+        }
+
+        return $filenames;
+    }
+
+    /**
+     * Set the path for the finding aid file.
+     *
+     * @param mixed $path
+     */
+    public function setPath($path)
+    {
+        $this->path = $path;
+    }
+
+    /**
+     * Get the relative path for the finding aid file.
+     *
+     * If $this->path has no value, check the list of possible finding aid file
+     * names for a matching file name on the filesystem.
+     *
+     * @return null|string path to finding aid file
+     */
+    public function getPath()
+    {
+        if (!isset($this->path)) {
+            foreach (self::getPossibleFilenames() as $filename) {
+                if (file_exists($this->getHomeDir().$filename)) {
+                    // Set relative path to file
+                    $this->path = $this->getHomeDir(false).$filename;
+                }
+            }
+        }
+
+        return $this->path;
+    }
+
+    /**
+     * Get the Finding Aid file format (pdf or rtf).
+     */
+    public function getFormat()
+    {
+        // If not set, try and determine the format from the file extension
+        if (!isset($this->format) && !empty($this->getPath())) {
+            $parts = pathinfo($this->getPath());
+
+            if (!empty($parts['extension'])) {
+                $this->format = strtolower($parts['extension']);
+            }
+        }
+
+        return $this->format;
+    }
+
+    /**
+     * Set the Finding Aid status.
+     *
+     * @param $status either self::GENERATED_STATUS or self::UPLOADED_STATUS
+     */
+    public function setStatus(int $status)
+    {
+        // Search for an existing 'findingAidStatus' property
```

Comment:
This propery setting code seems the same as around line 341... maybe it could be a helper function.

Reply:
Yeah, good call. :)